check-diff.sh

```diff
@@ -69,10 +69,14 @@ function set_environment_variables {
 			DIFF_BASE=${DIFF_BASE:-$TRAVIS_COMMIT^}
 		fi
 		DIFF_HEAD=${DIFF_HEAD:-$TRAVIS_COMMIT}
+	elif [[ ! -z "${BITBUCKET_BRANCH}" ]]; then
+		DIFF_BASE=${DIFF_BASE:-$BITBUCKET_COMMIT^}
```

Comment:
@glewiswp note that this will only compare with the previous commit which will lead to misleading results. See #197 and #229 for the same problem and a a proposed workaround for the same issue in Travis branch builds. What is needed here is the branch that the user intends to merge the changes into in order to be able to get the proper `$DIFF_BASE`. On Travis CI pull requests, this is available as `$TRAVIS_BRANCH` where it is defined as being the _destination_ branch for where the changes will be merged. Nevertheless, for BitBucket it seems that the `$BITBUCKET_BRANCH` variable is not the pull request destination branch but rather the _current_ branch, and thus it will have the exact same problem as Travis CI branch builds.